moped-editor/src/views/projects/projectView/SubprojectFundingModal.js

```diff
@@ -0,0 +1,169 @@
+    `https://data.austintexas.gov/resource/jega-nqf6.json?dept_unit_status=Active&sp_number_txt=${eCaprisID}&$limit=9999`
+  );
+
+  const [selectedFdus, setSelectedFdus] = useState([]);
+
+  // Filter the list of fdus to remove one(s) already on funding sources table
+  const filteredData = data.filter((fdu) => !fdusArray.includes(fdu.fdu));
+
+  const columns = [
+    {
+      field: "fdu",
+      title: "FDU",
+      cellStyle: { padding: "12px" },
+    },
+    {
+      field: "unit_long_name",
+      title: "Unit name",
+      cellStyle: { padding: "12px" },
+    },
+  ];
+
+  const handleAddFunding = () => {
+    const newFunds = [];
+    // format record to match generic records added
+    selectedFdus.forEach((fdu) => {
+      const fduRecord = {};
+      fduRecord.dept_unit = {
+        dept: fdu.dept,
+        dept_id: fdu.dept_id,
+        dept_unit_id: fdu.dept_unit_id,
+        dept_unit_status: fdu.dept_unit_status,
+        unit: fdu.unit,
+        unit_long_name: fdu.unit_long_name,
+        unit_short_name: fdu.unit_short_name,
+      };
+      fduRecord.fund = {
+        fund_id: fdu.fund,
+        fund_name: fdu.fundname,
+      };
+      fduRecord.project_id = projectId;
+      fduRecord.added_by = userId;
+      // funding status 2 is "Confirmed"
+      fduRecord.funding_status_id = 2;
+      newFunds.push(fduRecord);
+    });
+
+    // include records in mutation
+    addProjectFunding({
+      variables: {
+        objects: newFunds,
+      },
+    })
+      .then(() => handleDialogClose())
+      .catch((error) => {
+        setSnackbarState({
+          open: true,
+          message: (
+            <span>
+              There was a problem adding funding. Error message: {error.message}
+            </span>
+          ),
+          severity: "error",
+        });
+      });
+    setSelectedFdus([]);
+  };
+
+  return (
+    <Dialog
+      open={isDialogOpen}
+      onClose={handleDialogClose}
+      fullWidth
+      maxWidth={"md"}
+    >
+      <DialogTitle disableTypography className={classes.dialogTitle}>
+        <h3>Subproject funding sources</h3>
+        <IconButton onClick={() => handleDialogClose()}>
+          <CloseIcon />
+        </IconButton>
+      </DialogTitle>
+      <DialogContent>
+        <MaterialTable
+          columns={columns}
+          data={filteredData}
+          localization={{
+            body: {
+              emptyDataSourceMessage: (
+                <Typography>No FDUs available</Typography>
+              ),
+            },
+          }}
+          options={{
+            ...(data.length < 11 && {
+              paging: false,
+            }),
+            search: false,
+            toolbar: false,
+            tableLayout: "fixed",
+            selection: true,
+            rowStyle: typographyStyle,
+            pageSize: 10,
+            showSelectAllCheckbox: false,
+          }}
+          onSelectionChange={(rows) => setSelectedFdus(rows)}
+        />
```

Comment:
when the modal opens, i'm getting a warning about unique row IDs that i think is coming from around here.
```
The table requires all rows to have an unique id property. A row was provided without id in the rows prop. To prevent the loss of state between renders, please provide an unique id for each row.
```
Is there a prop you can pass somewhere to mute that?

Reply:
Ive seen that warning pop up on some of our other tables too, material table core docs says they have an option called "idSynonym" that will fix it, but it wasnt introduced until post v4.3.30. we are on 3.0.17, i want to check what breaking changes there are from v3 to v4

Reply:
👍 i wonder why it happens on some, but not all tables. is material-table expecting every row object to have some prop that it always checks for?

Reply:
it defaults to using "id" as the property, and some of our tables have that as a field, but not all